docs/source/devguide/styleguide.rst

```diff
@@ -158,6 +163,118 @@ each side.
 
 Do not leave trailing whitespace at the end of a line.
 
+---
+C++
+---
+
+Miscellaneous
+-------------
+
+Follow the `C++ Core Guidelines`_ except when they conflict with another
+guideline listed here.
+
+Conform to the C++11 standard.
+
+Always use C++-style comments (``//``) as opposed to C-style (``/**/``). (It
+is more difficult to comment out a large section of code that uses C-style
+comments.)
+
+Header files should always use include guards with the following style:
+
+.. code-block:: C++
+
+    #ifndef MODULE_NAME_H
+    #define MODULE_NAME_H
+    ...
+    content
+    ...
+    #endif // MODULE_NAME_H
+
+Do not use using-directives e.g. ``using namespace foobar;``
```

Comment:
I personally think we can leave this out. The core guidelines already have a good set of practices (SF.6 and SF.7).